File: service-results/php/ExchangeResultRequest.php
<?php

namespace Trinsic\ServiceResults;

/**
 * Standard request model for exchanging session result access keys
 */
class ExchangeResultRequest
{
    private string $sessionId;
    private string $resultsAccessKey;

    /**
     * Constructor
     * 
     * @param string $sessionId The unique identifier for the verification session
     * @param string $resultsAccessKey The access key needed to retrieve session results
     */
    public function __construct(string $sessionId, string $resultsAccessKey)
    {
        $this->sessionId = $sessionId;
        $this->resultsAccessKey = $resultsAccessKey;
    }

    /**
     * Gets the session ID
     * 
     * @return string
     */
    public function getSessionId(): string
    {
        return $this->sessionId;
    }

    /**
     * Sets the session ID
     * 
     * @param string $sessionId
     */
    public function setSessionId(string $sessionId): void
    {
        $this->sessionId = $sessionId;
    }

    /**
     * Gets the results access key
     * 
     * @return string
     */
    public function getResultsAccessKey(): string
    {
        return $this->resultsAccessKey;
    }

    /**
     * Sets the results access key
     * 
     * @param string $resultsAccessKey
     */
    public function setResultsAccessKey(string $resultsAccessKey): void
    {
        $this->resultsAccessKey = $resultsAccessKey;
    }

    /**
     * Validates the request
     * 
     * @throws ValidationException if validation fails
     */
    public function validate(): void
    {
        if (empty(trim($this->sessionId))) {
            throw new ValidationException('SessionId cannot be empty');
        }

        if (empty(trim($this->resultsAccessKey))) {
            throw new ValidationException('ResultsAccessKey cannot be empty');
        }

        if (!$this->isValidUuid($this->sessionId)) {
            throw new ValidationException('SessionId must be a valid UUID');
        }

        if (strlen($this->resultsAccessKey) < 10) {
            throw new ValidationException('ResultsAccessKey appears to be too short');
        }
    }

    /**
     * Checks if a string is a valid UUID
     * 
     * @param string $uuid
     * @return bool
     */
    private function isValidUuid(string $uuid): bool
    {
        return preg_match('/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i', $uuid) === 1;
    }

    /**
     * Converts the request to an array
     * 
     * @return array
     */
    public function toArray(): array
    {
        return [
            'sessionId' => $this->sessionId,
            'resultsAccessKey' => $this->resultsAccessKey,
        ];
    }

    /**
     * Creates an instance from an array
     * 
     * @param array $data
     * @return self
     * @throws ValidationException if required fields are missing
     */
    public static function fromArray(array $data): self
    {
        if (!isset($data['sessionId'])) {
            throw new ValidationException('sessionId is required');
        }

        if (!isset($data['resultsAccessKey'])) {
            throw new ValidationException('resultsAccessKey is required');
        }

        return new self($data['sessionId'], $data['resultsAccessKey']);
    }

    /**
     * String representation (with redacted access key)
     * 
     * @return string
     */
    public function __toString(): string
    {
        return sprintf('ExchangeResultRequest{sessionId: %s, resultsAccessKey: [REDACTED]}', $this->sessionId);
    }
}

File: service-results/python/exchange_result_request.py
"""
Standard request model for exchanging session result access keys
"""

import re
import uuid
from dataclasses import dataclass
from typing import Dict, Any


@dataclass
class ExchangeResultRequest:
    """Standard request model for exchanging session result access keys"""
    
    session_id: str
    results_access_key: str
    
    def __post_init__(self):
        """Validate the request after initialization"""
        self.validate()
    
    def validate(self) -> None:
        """
        Validates the ExchangeResultRequest
        
        Raises:
            ValueError: If validation fails
        """
        if not self.session_id or not self.session_id.strip():
            raise ValueError("session_id cannot be empty")
        
        if not self.results_access_key or not self.results_access_key.strip():
            raise ValueError("results_access_key cannot be empty")
        
        if not self._is_valid_uuid(self.session_id):
            raise ValueError("session_id must be a valid UUID")
        
        if len(self.results_access_key) < 10:
            raise ValueError("results_access_key appears to be too short")
    
    def _is_valid_uuid(self, uuid_string: str) -> bool:
        """
        Checks if a string is a valid UUID
        
        Args:
            uuid_string: The string to check
            
        Returns:
            True if valid UUID, False otherwise
        """
        try:
            uuid.UUID(uuid_string)
            return True
        except ValueError:
            return False
    
    def to_dict(self) -> Dict[str, str]:
        """
        Converts the request to a dictionary
        
        Returns:
            Dictionary representation of the request
        """
        return {
            "sessionId": self.session_id,
            "resultsAccessKey": self.results_access_key
        }
    
    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> 'ExchangeResultRequest':
        """
        Creates an ExchangeResultRequest from a dictionary
        
        Args:
            data: Dictionary containing the request data
            
        Returns:
            ExchangeResultRequest instance
            
        Raises:
            KeyError: If required fields are missing
            ValueError: If validation fails
        """
        try:
            session_id = data["sessionId"]
            results_access_key = data["resultsAccessKey"]
        except KeyError as e:
            raise KeyError(f"Missing required field: {e}")
        
        return cls(session_id=session_id, results_access_key=results_access_key)
    
    def sanitize_for_logging(self) -> Dict[str, str]:
        """
        Returns a sanitized version of the request for logging
        
        Returns:
            Dictionary with sensitive data redacted
        """
        return {
            "sessionId": self.session_id,
            "resultsAccessKey": "[REDACTED]"
        }
    
    def __str__(self) -> str:
        """String representation with redacted access key"""
        return f"ExchangeResultRequest(session_id={self.session_id}, results_access_key=[REDACTED])"